inst/htmljs/ldavis.js

```diff
@@ -1343,15 +1360,87 @@ LDAvis = function(to_select, json_file) {
         }
 
         function state_reset() {
+            
+            // set the style of any clicked term back to be non-underline                    
+            var old_term_clicked_id = termID + vis_state.term_clicked;
+            if (vis_state.term_clicked != "") {
+                var oldterm = document.getElementById(old_term_clicked_id);
+                if (oldterm != null) {
+                    oldterm.style.textDecoration = null;  
+                }
+            }
+                
             if (vis_state.topic > 0) {
                 topic_off(document.getElementById(topicID + vis_state.topic));
+                // set the style of any topic clicked to be back to regular style
+                // (no thick border around topic circle)
+                var old_topic_clicked_id = topicID + vis_state.topic_clicked;
+                if (vis_state.topic_clicked > 0) {
+                    document.getElementById(old_topic_clicked_id).style.strokeWidth = null;
+                }
             }
             if (vis_state.term != "") {
                 term_off(document.getElementById(termID + vis_state.term));
             }
             vis_state.term = "";
             document.getElementById(topicID).value = vis_state.topic = 0;
             state_save(true);
+            
+            // make sure term ids are all correct
+            d3.selectAll(".terms").attr("id", function(d) {
+              return (termID + d.Term)
+            });
+            
+            // update shiny inputs to be null
+            Shiny.onInputChange('ldavis_topic_clicked', null);
+            Shiny.onInputChange('ldavis_term_clicked', null);
+        }
```

Comment:
Have you tested this functionality outside of a shiny context? I think this would throw an error since `Shiny` is not defined if it's not running in a shiny app. In that case, we should add a check to see if we're in shiny mode similar to how HTMLwidgets does it -- https://github.com/ramnathv/htmlwidgets/blob/master/inst/www/htmlwidgets.js#L15

Reply:
I haven't..I'll add a check like the one you describe before calling `onInputChange()`.